unixfs/src/dir/builder/iter.rs

```diff
@@ -0,0 +1,416 @@
+                Ok(())
+            }
+        }
+
+        struct WriteableCid<'a>(&'a Cid);
+
+        // Cid by default does not have a way to count it's length or just write it out without
+        // allocating a vector.
+        impl<'a> MessageWrite for WriteableCid<'a> {
+            fn get_size(&self) -> usize {
+                use cid::Version::*;
+                use quick_protobuf::sizeofs::*;
+
+                match self.0.version() {
+                    V0 => self.0.hash().as_bytes().len(),
+                    V1 => {
+                        let version_len = 1;
+                        let codec_len = sizeof_varint(u64::from(self.0.codec()));
+                        let hash_len = self.0.hash().as_bytes().len();
+                        version_len + codec_len + hash_len
+                    }
+                }
+            }
+
+            fn write_message<W: WriterBackend>(
+                &self,
+                w: &mut Writer<W>,
+            ) -> quick_protobuf::Result<()> {
+                use cid::Version::*;
+
+                match self.0.version() {
+                    V0 => {
+                        for b in self.0.hash().as_bytes() {
+                            w.write_u8(*b)?;
+                        }
+                        Ok(())
+                    }
+                    V1 => {
+                        // it is possible that Cidv1 should not be linked to from a unixfs
+                        // directory; at least go-ipfs 0.5 `ipfs files` denies making a cbor link
+                        w.write_u8(1)?;
+                        w.write_varint(u64::from(self.0.codec()))?;
+                        for b in self.0.hash().as_bytes() {
+                            w.write_u8(*b)?;
+                        }
+                        Ok(())
+                    }
+                }
+            }
+        }
+
+        impl<'a> MessageWrite for BTreeMappedDir<'a> {
+            fn get_size(&self) -> usize {
+                use quick_protobuf::sizeofs::*;
+
+                let links = self
+                    .links
+                    .iter()
+                    .map(|(k, v)| EntryAsPBLink(k, v))
+                    .map(|link| 1 + sizeof_len(link.get_size()))
+                    .sum::<usize>();
+
+                links + 1 + sizeof_len(self.data.get_size())
+            }
+            fn write_message<W: WriterBackend>(
+                &self,
+                w: &mut Writer<W>,
+            ) -> quick_protobuf::Result<()> {
+                for l in self.links.iter().map(|(k, v)| EntryAsPBLink(k, v)) {
+                    w.write_with_tag(18, |w| w.write_message(&l))?;
+                }
+                w.write_with_tag(10, |w| w.write_message(&self.data))
+            }
+        }
+
+        let btreed = BTreeMappedDir {
+            links,
+            data: UnixFs {
+                Type: UnixFsType::Directory,
+                Data: None,
```

Comment:
```suggestion
```

Reply:
I was thinking this is more like a comment that there is no Data but probably someone could dig it up through the default as well.